src/Core/OrganizationFeatures/OrganizationSubscriptions/AddSecretsManagerSubscriptionCommand.cs

```diff
@@ -0,0 +1,154 @@
+    {
+        if (organization == null)
+        {
+            throw new NotFoundException();
+        }
+
+        if (string.IsNullOrWhiteSpace(organization.GatewayCustomerId))
+        {
+            throw new GatewayException("Not a gateway customer.");
+        }
+
+        if (string.IsNullOrWhiteSpace(organization.GatewaySubscriptionId))
+        {
+            throw new BadRequestException("No subscription found.");
+        }
+    }
+
+    private void ValidateSecretsManagerPlan(Plan plan, Organization signup, int additionalSeats, int additionalServiceAccounts)
+    {
+        if (plan is not { LegacyYear: null })
+        {
+            throw new BadRequestException($"Invalid Secrets Manager plan selected.");
+        }
+
+        if (plan.Disabled)
+        {
+            throw new BadRequestException($"Secrets Manager Plan is disabled.");
+        }
+
+        if (plan.BaseSeats + additionalSeats <= 0)
+        {
+            throw new BadRequestException($"You do not have any Secrets Manager seats!");
+        }
+
+        if (additionalSeats < 0)
+        {
+            throw new BadRequestException($"You can't subtract Secrets Manager seats!");
+        }
+
+        if (!plan.HasAdditionalServiceAccountOption && additionalServiceAccounts > 0)
+        {
+            throw new BadRequestException("Plan does not allow additional Service Accounts.");
+        }
+
+        if (additionalSeats > signup.Seats)
+        {
+            throw new BadRequestException("You cannot have more Secrets Manager seats than Password Manager seats.");
+        }
+
+        if (additionalServiceAccounts < 0)
+        {
+            throw new BadRequestException("You can't subtract Service Accounts!");
+        }
+
+        switch (plan.HasAdditionalSeatsOption)
+        {
+            case false when additionalSeats > 0:
+                throw new BadRequestException("Plan does not allow additional users.");
+            case true when plan.MaxAdditionalSeats.HasValue &&
+                           additionalSeats > plan.MaxAdditionalSeats.Value:
+                throw new BadRequestException($"Selected plan allows a maximum of " +
+                                              $"{plan.MaxAdditionalSeats.GetValueOrDefault(0)} additional users.");
+        }
+    }
+
+    private async Task<Organization> SignUpAsync(Organization organization)
+    {
+        try
+        {
+            await _organizationService.ReplaceAndUpdateCacheAsync(organization);
+
+            var ownerUsers =
+                await _organizationUserRepository.GetManyByOrganizationAsync(organization.Id,
+                    OrganizationUserType.Owner);
+
+            if (ownerUsers.Any())
+            {
+                foreach (var onwerUser in ownerUsers)
+                {
+                    onwerUser.AccessSecretsManager = true;
+                }
+
+                await _organizationUserRepository.ReplaceManyAsync(ownerUsers);
+            }
```

Comment:
We can't assume this. There might be more owners than SmSeats, or maybe not every owner needs SM access.  I would delete this part and let the owner manually provision SM access.

You can then delete this private method and inline any remaining code into the main method.

Reply:
done